### PR TITLE
Finalize handling of IPlaceholderOperation in control flow graph.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review

File: src/Compilers/Core/Portable/Operations/OperationVisitor.cs [hunk 1]
```diff
@@ -145,6 +145,12 @@ internal virtual void VisitFixed(IFixedOperation operation)
             VisitNoneOperation(operation);
         }
 
+        internal virtual void VisitAggregateQuery(IAggregateQueryOperation operation)
+        {
+            //DefaultVisit(operation);
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Operations/OperationVisitor.cs]: Can the comment be removed (here and below, at line 709)?

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        <Fact()>
+        Public Sub AggregateFlow_02()
+            Dim source = <![CDATA[
+Imports System
+Imports System.Linq
+
+Public Structure C
+
+    Sub M(a As Integer(), b As Integer(), result As object) 'BIND:"Sub M"
+        result = Aggregate y In If(a, b) Into Count(), Sum(y)
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb]: Does the `Into` syntax allow for explicit names for the anonymous type elements?

**Review by jcouv — APPROVED** [review]
LGTM

**Comment by AlekseyTs** [comment]
@333fred, @cston, @dotnet/roslyn-compiler Please review, need second review

**Comment by AlekseyTs** [comment]
@mavasani, @dotnet/analyzer-ioperation Please review

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs [hunk 3]
```diff
@@ -5593,6 +5607,40 @@ public override IOperation VisitOmittedArgument(IOmittedArgumentOperation operat
             return new OmittedArgumentExpression(semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, IsImplicit(operation));
         }
 
+        internal override IOperation VisitPlaceholder(IPlaceholderOperation operation, int? captureIdForResult)
```
Comment (agocke) [inline src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: Why did this method move?

**Review by agocke — APPROVED** [review]
LGTM

File: src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs (hunk 3 above)
Comment (AlekseyTs) [reply src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs]: > Why did this method move?

The region below contains methods that haven't been finalized. Once that happens, we move the method out of it.

**Review by mavasani — APPROVED** [review]

File: src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb (hunk 2 above)
Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb]: > Does the Into syntax allow for explicit names for the anonymous type elements?

Yes it does.

---
In reply to: [192755825](https://github.com/dotnet/roslyn/pull/27391#discussion_r192755825) [](ancestors = 192755825)

Comment (jcouv) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb]: Would it make sense to test an aggregate with names then?

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ITranslatedQueryOperation.vb]: > Would it make sense to test an aggregate with names then?

Names play no role in the control flow or the shape of the nodes we are going after in this PR. Given the time constraints, we are not doing discovery testing.

---
In reply to: [192882767](https://github.com/dotnet/roslyn/pull/27391#discussion_r192882767) [](ancestors = 192882767)

**Pull request merged**